[EuiDataGrid][BugFix]: Fix visible overflowing truncated text (#7793)
elastic/eui: packages/eui/changelogs/upcoming/7793.md

**Bug fixes**

> - Fixed a visual bug on `EuiDataGrid` cells when using `rowHeightOptions.defaultHeight.lineCount` where the clamped text was still visible for some font sizes

Reviewer: Changelog copy tweak
```suggestion
- Fixed an `EuiDataGrid` visual bug when using `lineCount` row heights where the clamped text was still visible for some font sizes
```
